sdk/core/AzureCore/Source/Pipeline/PipelineClient.swift

```diff
@@ -36,6 +36,8 @@ public protocol AzureClientOptions {
     var telemetryOptions: TelemetryOptions { get }
     /// Global transport options
     var transportOptions: TransportOptions { get }
+    /// The default dispatch queue on which to call all completion handlers. Defaults to `DispatchQueue.main`.
+    var dispatchQueue: DispatchQueue? { get }
```

Comment:
This doesn't seem to be true. I only see the `dispatchQueue` property being used for the PagedCollection (where it defaults to `.utility`), and the other client operations don't appear to use the dispatchQueue at all anymore.

Reply:
The generated client code in autorest.swift uses:
```Swift
let dispatchQueue = self.commonOptions.dispatchQueue ?? DispatchQueue.main
dispatchQueue.async { 
  stuff
}
```

I removed any references to dispatch queues inside the collections file because those calls were not working. They were hanging even though `commonOptions.dispatchQueue` was set to the right value (and thus these were being called on the right queue). Removing any reference to the dispatchQueue fixed the issue and still permits the final completion handler call on the desired queue.

Reply:
Here's the autorest.swift PR link that shows how this change is translating to generated code:
https://github.com/Azure/autorest.swift/pull/177